### PR TITLE
Webhook: Fix JIRA key error for name

### PR DESCRIPTION
**Pull request by madchap**
Addresses #3731

### CODE REVIEW

**Comment by valentijnscholten** [comment]
Not sure if you should split the email address. Probably the jira username Used by defect dojo is also an email address in that case so need to compare against that. Or against both maybe.

**Comment by madchap** [comment]
You're right, just made the test, email is in full. Will amend.

**Comment by madchap** [comment]
I was looking at the fixture we use for test and I could not see easily which jira configuration would be used for whatever jira_id is specified in the unittest. Seems no product actually has it? Maybe I missed it, will look again tomorrow.

**Comment by valentijnscholten** [comment]
what we are seeing is that some instances send only name some instance only emailaddress. So two separate cases, needing 2 separate code paths / unit tests? You can use finding5 for bothI think, but your new testcase shouldn't have the name field in the json I think, only email.

File: dojo/fixtures/dojo_testdata.json [hunk 1]
```diff
@@ -1686,6 +1686,24 @@
...
+      "url": "http://www.testjira.com",
+      "medium_mapping_severity": "test severity",
+      "open_status_key": 333,
+      "close_status_key": 334,
+      "critical_mapping_severity": "test severity",
+      "high_mapping_severity": "test severity",
+      "password": "user69",
+      "epic_name_id": 666
+    }
+  },
```
Comment (valentijnscholten) [inline dojo/fixtures/dojo_testdata.json:1706]: do we need this?

Comment (valentijnscholten) [reply dojo/fixtures/dojo_testdata.json:1706]: ah yes to make one with an email address.

Comment (madchap) [reply dojo/fixtures/dojo_testdata.json:1706]: yea, just have to fiddle with extra things now in the fixture...

Comment (valentijnscholten) [reply dojo/fixtures/dojo_testdata.json:1706]: easier just to change the username on jira_instance with id 2 in python code and do .save()

Comment (valentijnscholten) [reply dojo/fixtures/dojo_testdata.json:1706]: fixtures are a pain and should onyl be used for a very minimal set of data to get the instance going.

**Comment by madchap** [comment]
Should be OK now, except for that unrelated report timeout selenium test failing intermittently.

**Review by valentijnscholten — APPROVED** [review]

**Comment by valentijnscholten** [comment]
thanks for fixing this!

**Review by damiencarol — APPROVED** [review]

**Pull request merged**